src/components/AssignVariantDialog/AssignVariantDialog.tsx

```diff
@@ -0,0 +1,222 @@
+  onSubmit: (data: SearchVariants_search_edges_node[]) => void;
+}
+
+function handleVariantAssign(
+  variant: SearchVariants_search_edges_node,
+  isSelected: boolean,
+  selectedVariants: SearchVariants_search_edges_node[],
+  setSelectedVariants: (data: SearchVariants_search_edges_node[]) => void
+) {
+  if (isSelected) {
+    setSelectedVariants(
+      selectedVariants.filter(
+        selectedVariant => selectedVariant.id !== variant.id
+      )
+    );
+  } else {
+    setSelectedVariants([...selectedVariants, variant]);
+  }
+}
+
+const scrollableTargetId = "assignVariantScrollableDialog";
+
+const AssignVariantDialog: React.FC<AssignVariantDialogProps> = props => {
+  const {
+    confirmButtonState,
+    hasMore,
+    open,
+    loading,
+    variants,
+    onClose,
+    onFetch,
+    onFetchMore,
+    onSubmit
+  } = props;
+  const classes = useStyles(props);
+  const scrollableDialogClasses = useScrollableDialogStyle({});
+
+  const intl = useIntl();
+  const [query, onQueryChange] = useSearchQuery(onFetch);
+  const [selectedVariants, setSelectedVariants] = React.useState<
+    SearchVariants_search_edges_node[]
+  >([]);
+
+  const handleSubmit = () => onSubmit(selectedVariants);
+
+  return (
+    <Dialog
+      onClose={onClose}
+      open={open}
+      classes={{ paper: scrollableDialogClasses.dialog }}
+      fullWidth
+      maxWidth="sm"
+    >
+      <DialogTitle>
+        <FormattedMessage
+          defaultMessage="Assign Variant"
+          description="dialog header"
+        />
+      </DialogTitle>
+      <DialogContent className={scrollableDialogClasses.topArea}>
+        <TextField
+          name="query"
+          value={query}
+          onChange={onQueryChange}
+          label={intl.formatMessage({
+            defaultMessage: "Search Variants"
+          })}
+          placeholder={intl.formatMessage({
+            defaultMessage:
+              "Search by variant name, attribute, variant type etc..."
+          })}
+          fullWidth
+          InputProps={{
+            autoComplete: "off",
+            endAdornment: loading && <CircularProgress size={16} />
+          }}
+        />
+      </DialogContent>
+      <DialogContent
+        className={scrollableDialogClasses.scrollArea}
+        id={scrollableTargetId}
+      >
+        <InfiniteScroll
+          dataLength={variants?.length}
+          next={onFetchMore}
+          hasMore={hasMore}
+          scrollThreshold="100px"
+          loader={
+            <div className={scrollableDialogClasses.loadMoreLoaderContainer}>
+              <CircularProgress size={16} />
+            </div>
+          }
+          scrollableTarget={scrollableTargetId}
+        >
+          <ResponsiveTable key="table">
+            <TableBody>
+              {variants &&
+                variants.map(variant => {
+                  const isSelected = selectedVariants.some(
+                    selectedVariant => selectedVariant.id === variant.id
```

Comment:
You mentioned above that we should use only id's, not whole variants, it's a good point, but after doing this i wold be able to use getById since it operates on objects. Same for `getByUmatchingId`.